File: trees/sameTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    void sol(TreeNode* p, vector<int>& ans1){
        if(p==NULL) {
            ans1.push_back(INT_MIN);
            return;
        }

        ans1.push_back(p->val);
        sol(p->left, ans1);
        sol(p->right, ans1);
    }
    
    
    bool isSameTree(TreeNode* p, TreeNode* q) {
    
        vector<int> ans1;
        vector<int> ans2;

        sol(p, ans1);
        sol(q, ans2);

        if(ans1.size()!=ans2.size()) return false;

        for(int i=0; i<ans1.size(); i++){
            if(ans1[i]!=ans2[i]) return false;
        }

        return true;
    
    }
};

File: trees/sameTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:

    def sol(self, n: TreeNode | None, arr: List[int]):
        if n==None:
            arr.append(float('-inf'))
            return

        arr.append(n.val)

        self.sol(n.left, arr)
        self.sol(n.right, arr)

    def isSameTree(self, p: TreeNode | None, q: TreeNode | None) -> bool:
        arr1=[]
        arr2=[]
        
        self.sol(p, arr1)
        self.sol(q, arr2)

        if len(arr1)!=len(arr2):
            return False

        # for i in range(len(arr1)):
        #     if arr1[i]!= arr2[i]:
        #         return False

        if arr1!=arr2:
            return False

        return True
